python/ray/__init__.py

```diff
@@ -284,7 +285,7 @@ def __getattr__(self, attr):
 def __getattr__(name: str):
     import importlib
 
-    if name in ["data", "workflow", "autoscaler"]:
+    if name in ["data", "workflow", "autoscaler", "train"]:
```

Comment:
what's this?

Reply:
Good point. I don't think they are necessary. There was a unit test failing. I think the `ray.train.setup.py` solved the problem. I will follow the discussion thread of @omatthew98 provided to remove the dependency of `yaml` by replacing it as a python file with yaml string constants. 

The changes here should be reverted.

Reply:
Without add `train` here. There will be some dependency error when calling `import ray.train`

Because we add `configure_log()` in `ray.train.__init__.py`. However, the configure_log function itself will import classes from ray train, which assumes `ray.train` is already there.

Please let me know if there are better solutions to solve this dependecy problem.